### DISCUSSION TITLE
Using the regex property for input validation

### DISCUSSION THREAD

**Discussion by mirai-cmd** [react-hook-form/react-hook-form#10221] [General] [answered]
Basically I wanted to use regex patterns to validate input fields. Specifically print the regex error message.
As of now I'm able to render a span with the "required" property error message. But I'm not able to use the regex error message. Here's my code fragment:
`{...register("fullName",{
     required: "This is a required field",
     pattern: {
         value: "/[A-Za-z]+/",
          message : "Name can only have letters !"
      }
})}`
I logged the errors object too, but the regex error is not coming up. I would like to use the pattern message in the span depending on the input. Kindly point out if I'm doing something wrong and if any additional details are required.

**Comment by Sajarin-M** [comment, marked as answer]
The regex you are using checks if the string has at least one alphabet. It doesn't restrict other characters from being present.
It produces an error if we type `1122` but not if we type `a1122` because the it contains an alphabet.

To check if a string has only alphabets from beginning to end, you can use `/^[a-zA-Z]+$/`.
This matches the string only if it contains alphabets and nothing else. 

Also, there is no need to wrap the regex in double quotes.

**Reply by mirai-cmd** [reply]
Thank you so much, I changed the regex it,worked. Appreciate it.
